src/formatDate.js

```diff
@@ -7,8 +7,37 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateString, fromFormat, toFormat) {
+  const separator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const dateParts = dateString.split(separator);
+
+  const dateObj = {
+    YYYY: fromFormat.includes('YYYY')
+      ? dateParts[fromFormat.indexOf('YYYY')]
+      : null,
+    YY: fromFormat.includes('YY') ? dateParts[fromFormat.indexOf('YY')] : null,
+    MM: dateParts[fromFormat.indexOf('MM')],
+    DD: dateParts[fromFormat.indexOf('DD')],
+  };
```

Comment:
Consider using a loop to create the dateObj. Since you are performing similar operations for each date part, a loop could make your code more concise and maintainable. For example, you could iterate over the `fromFormat` array (excluding the separator) and build the `dateObj` dynamically.